File: app/src/main/java/ua/syt0r/furiganit/model/repository/overlay/OverlayDataRepository.java
package ua.syt0r.furiganit.model.repository.overlay;

import android.content.Context;
import android.content.SharedPreferences;
import android.graphics.Point;

/*
* Saves and restores position of overlay window
* */
public class OverlayDataRepository {

    private static final String PREF_NAME = "overlay_window_pos";
    private static final String X_POS = "xPos";
    private static final String Y_POS = "yPos";
    private static final String TIMEOUT = "timeout";

    private SharedPreferences sharedPreferences;

    public OverlayDataRepository(Context context) {
        sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }

    public Point getPosition() {
        int x = sharedPreferences.getInt(X_POS, 0);
        int y = sharedPreferences.getInt(Y_POS, 0);
        return new Point(x, y);
    }

    public void setPosition(int xPos, int yPos) {
        sharedPreferences.edit()
                .putInt(X_POS, xPos)
                .putInt(Y_POS, yPos)
                .apply();
    }

    public long getTimeout() {
        return sharedPreferences.getLong("timeout",5);
    }

    public void setTimeout(long value) {
        sharedPreferences.edit().putLong(TIMEOUT, value).apply();
    }


}

File: app/src/main/java/ua/syt0r/furiganit/model/repository/overlay/OverlayDataRepository.kt
package ua.syt0r.furiganit.model.repository.overlay

import android.content.Context
import android.content.SharedPreferences
import android.graphics.Point

/*
* Saves and restores position of overlay window
*/
class OverlayDataRepository(context: Context) {

    private val sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)

    var position: Point
        get() {
            val x = sharedPreferences.getInt(X_POS, 0)
            val y = sharedPreferences.getInt(Y_POS, 0)
            return Point(x, y)
        }
        set(value) = sharedPreferences.edit()
            .putInt(X_POS, value.x)
            .putInt(Y_POS, value.y)
            .apply()

    var timeout: Long
        get() = sharedPreferences.getLong("timeout", 5)
        set(value) = sharedPreferences.edit().putLong(TIMEOUT, value).apply()

    companion object {
        private const val PREF_NAME = "overlay_window_pos"
        private const val X_POS = "xPos"
        private const val Y_POS = "yPos"
        private const val TIMEOUT = "timeout"
    }

}
